### ISSUE TITLE
prismaSchemaFolder don't work properly

### ISSUE THREAD

**Issue by poyrazhancilar** [prisma/prisma#25607]
### Discussed in https://github.com/prisma/prisma/discussions/25606

<div type='discussions-op-text'>

<sup>Originally posted by **poyrazhancilar** November  8, 2024</sup>
### Question

I am trying to use multiple Prisma files with Prisma, and I have reviewed everything, including Prisma's own documentation. However, whenever I run npx prisma generate, I get the following error:

```bash
> npx prisma generate
Environment variables loaded from .env
Prisma schema loaded from src/prisma/schema/schema.prisma
Error: 
You don't have any models defined in your schema.prisma, so nothing will be generated.
You can define a model like this:

model User {
  id    String  @id @default(auto()) @map("_id") @db.ObjectId
  email String  @unique
  name  String?
}

More information in our documentation:
https://pris.ly/d/prisma-schema
```

File & Folder 
<img width="294" alt="Screenshot 2024-11-08 at 15 51 27" src="https://github.com/user-attachments/assets/8ac9740b-43e0-4db8-ac36-0022a53dec5f">

Structure:

schema.prisma:
```prisma
datasource db_main {
  provider = "mongodb"
  url      = env("MONGODB_URI")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["prismaSchemaFolder"]
}
```

contact.prisma
```prisma
datasource db_contact {
  provider = "mongodb"
  url      = env("MONGODB_URI")
}

model ContactRequest {
  id          String  @id @default(cuid()) @map("_id")
  email       String
  message String?

  @@map("contact_requests")
}
```

Why am I getting this error? I have checked the things in the documentation and other sources, but I am still experiencing this issue. I would appreciate it if you could help, thank you.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it
</div>

**Comment by nayeemuddin-bd100** [comment]
facing the same issue

**Comment by aqrln** [comment]
Hey @poyrazhancilar, I don't see anything wrong immediately (other than two datasources but that should lead to this problem specifically). Could you please provide a clonable reproduction as a GitHub repo, as well as the missing information from the issue template?

**Comment by clemens** [comment]
Unless I'm wrong, the main `schema.prisma` still needs to be in the main `prisma` folder, no? So `prisma/schema.prisma` and `prisma/schema/xyz.prisma`.

**Comment by jacek-prisma** [comment]
@poyrazhancilar, it seems like your main schema is in the wrong location, could you retry with corrected location, as outlined in the docs: https://www.prisma.io/docs/orm/prisma-schema/overview/location#prisma-schema-location

**Comment by jacek-prisma** [comment]
Closing since there's been no feedback for a week.

**Comment by storyd** [comment]
So I'm encountering a similar issue involving AWS CDK esbuild process. I will attempt to cover my discovery, but I have to redact certain information and due to the CDK abstractions before prisma specific processes, making a simplified repo might be problematic.

from the CDK in a class extended from `aws-cdk-lib/aws-lambda-nodejs/NodejsFunction`
`props.bundling.commandHooks.afterBundling` we have the output of a few debug commands

```
# ls -laR /workspaces/REDACTED/cdk.out/bundling-temp-5cde0b06645d967d92b27f28f951454b95527f3a827cefe9d4fc82e17c74a23f/prisma
/workspaces/REDACTED/cdk.out/bundling-temp-5cde0b06645d967d92b27f28f951454b95527f3a827cefe9d4fc82e17c74a23f/prisma:
total 24
drwxr-xr-x+ 4 codespace codespace 4096 Jan 29 00:16 .
drwxrwxrwx+ 4 codespace codespace 4096 Jan 29 00:16 ..
-rw-rw-rw-  1 codespace codespace  313 Jan 29 00:16 .env
drwxr-xr-x+ 3 codespace codespace 4096 Jan 29 00:16 migrations
drwxr-xr-x+ 2 codespace codespace 4096 Jan 29 00:16 schema
-rw-rw-rw-  1 codespace codespace  275 Jan 29 00:16 seed.ts

/workspaces/REDACTED/cdk.out/bundling-temp-5cde0b06645d967d92b27f28f951454b95527f3a827cefe9d4fc82e17c74a23f/prisma/migrations:
total 16
drwxr-xr-x+ 3 codespace codespace 4096 Jan 29 00:16 .
drwxr-xr-x+ 4 codespace codespace 4096 Jan 29 00:16 ..
drwxr-xr-x+ 2 codespace codespace 4096 Jan 29 00:16 initial
-rw-rw-rw-  1 codespace codespace  127 Jan 29 00:16 migration_lock.toml

/workspaces/REDACTED/cdk.out/bundling-temp-5cde0b06645d967d92b27f28f951454b95527f3a827cefe9d4fc82e17c74a23f/prisma/migrations/initial:
total 16
drwxr-xr-x+ 2 codespace codespace 4096 Jan 29 00:16 .
drwxr-xr-x+ 3 codespace codespace 4096 Jan 29 00:16 ..
-rw-rw-rw-  1 codespace codespace 6117 Jan 29 00:16 migration.sql

/workspaces/REDACTED/cdk.out/bundling-temp-5cde0b06645d967d92b27f28f951454b95527f3a827cefe9d4fc82e17c74a23f/prisma/schema:
total 48
drwxr-xr-x+ 2 codespace codespace 4096 Jan 29 00:16 .
drwxr-xr-x+ 4 codespace codespace 4096 Jan 29 00:16 ..
-rw-rw-rw-  1 codespace codespace  739 Jan 29 00:16 REDACTED.prisma
-rw-rw-rw-  1 codespace codespace  361 Jan 29 00:16 REDACTED.prisma
-rw-rw-rw-  1 codespace codespace  741 Jan 29 00:16 REDACTED.prisma
-rw-rw-rw-  1 codespace codespace  431 Jan 29 00:16 REDACTED.prisma
-rw-rw-rw-  1 codespace codespace  401 Jan 29 00:16 REDACTED.prisma
-rw-rw-rw-  1 codespace codespace   70 Jan 29 00:16 REDACTED.prisma
-rw-rw-rw-  1 codespace codespace  474 Jan 29 00:16 schema.prisma
-rw-rw-rw-  1 codespace codespace  384 Jan 29 00:16 REDACTED.prisma
-rw-rw-rw-  1 codespace codespace  278 Jan 29 00:16 REDACTED.prisma
-rw-rw-rw-  1 codespace codespace 1030 Jan 29 00:16 REDACTED.prisma
```

```
# cat /workspaces/REDACTED/cdk.out/bundling-temp-5cde0b06645d967d92b27f28f951454b95527f3a827cefe9d4fc82e17c74a23f/package.json
# note, this is generated by the CDK and access to fields other than dependencies is problematic at best.
{"dependencies":{"pg":"^8.13.1","@prisma/adapter-pg":"^6.1.0","prisma":"^6.1.0","@prisma/client":"^6.1.0","@aws-sdk/rds-signer":"^3.716.0"}}
```

```
# cat /workspaces/REDACTED/cdk.out/bundling-temp-5cde0b06645d967d92b27f28f951454b95527f3a827cefe9d4fc82e17c74a23f/prisma/schema/schema.prisma
generator client {
  provider        = "prisma-client-js"
  // make cold start fast especially for large schema
  // https://github.com/prisma/prisma/issues/18095
  previewFeatures = ["postgresqlExtensions", "driverAdapters", "prismaSchemaFolder", "omitApi", "strictUndefinedChecks"]
  binaryTargets = ["native", "rhel-openssl-3.0.x"]
}

datasource db {
  provider          = "postgresql"
  url               = env("DATABASE_URL")
  extensions        = [pg_trgm, postgis]
}
```

without diving into redacted details, the models in the schema folder have content, match the src schema files and work as intended during `npx prisma generate` outside the CDK SYNTH environment.

at this point, the following command generates errors
```
# npx prisma generate --schema=.../prisma/schema/schema.prisma
# note: ... in above used for post brevity, path is correct
Environment variables loaded from prisma/.env
Prisma schema loaded from prisma/schema/schema.prisma
Error: 
You don't have any models defined in your schema.prisma, so nothing will be generated.
...
```

this has been a VERY long roadblock in using prisma for this project trying to get a working esbuild with the aws cdk working.
we are in this bundling cli `prisma generate` line of debugging as we were following a post where we needed to reference some deep classes did not resolve a runtime error of `"Error: @prisma/client did not initialize yet. Please run \"prisma generate\" and try to import it again."`

```
// HACK: esbuild bundling workaround
// See: https://www.prisma.io/docs/orm/prisma-client/deployment/serverless/deploy-to-aws-lambda
// @ts-ignore
import schema from './prisma/schema/schema.prisma';
// @ts-ignore
import x from './node_modules/.prisma/client/libquery_engine-rhel-openssl-3.0.x.so.node'
```

in the output of the bundle, this is in the `node_modules/.prisma/client.index.js` and **does not** match the outside of CDK `node_modules/.prisma/client.index.js` file post generate.
```
// src/scripts/default-index.ts
var PrismaClient = class {
  constructor() {
    throw new Error('@prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.');
  }
};
```

**Comment by storyd** [comment]
Our solution was subtle and easy to overlook. Our project is complex and requires us to break out of the prisma box of doing thing in several places; luckily, prisma offers these overrides. Our issue was where these overrides converged and the error message was vague.

from the documentation on [multiple Prisma schema files](https://www.prisma.io/docs/orm/prisma-schema/overview/location#how-to-use-existing-prisma-cli-commands-with-multiple-prisma-schema-files)
`npx prisma db push --schema custom/path/to/my/schema  # note this is now a directory!`

to simplify our project, we consolidated the seventeen (17!) references to the various ways to denote `my schema location is not in the usual place` to a generator script when we initially started the conversion to prisma, at that time we were using the monolithic `schema.prisma` file.

our `package.json` when the prisma utilities and libraries are executed is generated, but for the simple project

**tl;dr;**

```diff
{
  "name": "REDACTED Build ...",
  "main": "index.js",
  "prisma": {
-    "schema": "src/prisma/schema/schema.prisma",
+    "schema": "src/prisma/schema/",
    ...
  }
  ...
}
```

**Comment by BOSSgua** [comment]
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/1
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/2
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/3
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/4
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/5
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/6
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/7
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/8
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/9
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/10
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/11
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/12
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/13
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/14
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/15
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/16
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/17
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/18
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/19
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/20
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/21
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/22
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/23
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/24
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/25
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/26
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/27
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/28
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/29
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/30
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/31
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/32
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/33
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/34
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/35
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/36
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/37
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/38
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/39
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/40
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/41
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/42
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/43
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/44
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/45
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/46
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/47
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/48
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/49
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/50
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/51
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/52
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/53
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/54
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/55
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/56
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/57
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/58
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/59
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/60
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/61
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/62
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/63
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/64
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/65
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/66
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/67
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/68
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/69
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/70
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/71
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/72
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/73
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/74
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/75
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/76
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/77
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/78
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/79
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/80
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/81
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/82
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/83
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/84
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/85
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/86
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/87
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/88
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/89
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/90
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/91
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/92
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/93
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/94
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/95
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/96
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/97
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/98
https://github.com/linbojuezhiming/linbojuezhiming.github.io/issues/99
